pkg/build/builder/cmd/dockercfg/cfg.go

```diff
@@ -57,16 +53,11 @@ func (h *Helper) GetDockerAuth(imageName, authType string) (docker.AuthConfigura
 		return docker.AuthConfiguration{}, false
 	}
 
-	var cfg credentialprovider.DockerConfig
-	var err error
-	if strings.HasSuffix(dockercfgPath, kapi.DockerConfigJsonKey) || strings.HasSuffix(dockercfgPath, "config.json") {
-		cfg, err = readDockerConfigJson(dockercfgPath)
-	} else if strings.HasSuffix(dockercfgPath, kapi.DockerConfigKey) {
-		cfg, err = readDockercfg(dockercfgPath)
-	}
-
+	dockercfgDir, dockercfgFile := filepath.Split(dockercfgPath)
+	credentialprovider.SetPreferredDockercfgPath(dockercfgDir)
```

Comment:
@bparees I add some change again, ptal,
```
k8s to improve the usability of the config reader function before trying to
leverage it here.
```
should we add one docker config parameter in function ReadDockerConfigFile?